Frends.AzureBlobStorage.ListBlobsInContainer/Frends.AzureBlobStorage.ListBlobsInContainer/ListBlobsInContainer.cs

```diff
@@ -0,0 +1,151 @@
+                var flatBlobListing = await FlatBlobListing(enumerator, source, blobsList, authSas);
+                return new Result { BlobList = flatBlobListing };
+
+            }
+            else
+            {
+                var enumerable = blobContainerClient.GetBlobsByHierarchyAsync(BlobTraits.None, BlobStates.None, "/", string.IsNullOrWhiteSpace(source.Prefix) ? null : source.Prefix).AsPages();
+                var enumerator = enumerable.GetAsyncEnumerator();
+                var listBlobsHierarchy = await ListBlobsHierarchy(enumerator, source, blobsList, authSas);
+                return new Result { BlobList = listBlobsHierarchy };
+            }
+        }
+
+
+        /// <summary>
+        ///     List blobs in a flat listing structure.
+        /// </summary>
+        /// <param name="source">Source values.</param>
+        /// <param name="blobsList">List of blobs in container.</param>
+        /// <param name="authSas">Authentication method.</param>
+        /// <returns>List of blobs in container with flat listing structure.</returns>
+        public static async Task<List<Result>> FlatBlobListing(IAsyncEnumerator<Page<BlobItem>> enumerator, Source source, List<Result> blobsList, bool authSas)
+        {
+            try
+            {
+                while (await enumerator.MoveNextAsync())
+                {
+                    var blobItems = enumerator.Current;
+                    foreach (var blobItem in blobItems.Values)
+                    {
+                        var blob = authSas ? new BlobClient(new Uri($"{source.Uri}/{source.ContainerName}/{blobItem.Name}?")) : new BlobClient(source.ConnectionString, source.ContainerName, blobItem.Name);
+                        blobsList.Add(new Result
+                        {
+                            BlobType = blobItem.Properties.BlobType.ToString(),
+                            Uri = blob.Uri.ToString(),
+                            Name = blob.Name,
+                            ETag = blobItem.Properties.ETag.ToString()
+                        });
+                    }
+                }
+            }
+            catch (UnauthorizedAccessException ex)
+            {
+                throw new Exception("Check authentication information." + ex.ToString());
+            }
+            finally
+            {
+                await enumerator.DisposeAsync();
+            }
+            return blobsList;
+        }
+
+        /// <summary>
+        ///  List blobs hierarchically.
+        /// </summary>
+        /// <param name="source">Source values.</param>
+        /// <param name="blobsList">List of blobs in container.</param>
+        /// <param name="authSas">Authentication method.</param>
+        /// <returns>List of blobs in container with hierarchical structure.</returns>
+        public static async Task<List<Result>> ListBlobsHierarchy(IAsyncEnumerator<Page<BlobHierarchyItem>> enumerator, Source source, List<Result> blobsList, bool authSas)
```

Comment:
This is a helper method, so it should be private. Also, you should pass cancellation token as a parameter.